api/graphql/applications/application_serializers.py

```diff
@@ -0,0 +1,247 @@
+        help_text="Contact person information for the application",
+        read_only=False,
+        allow_null=True,
+    )
+
+    organisation = OrganisationCreateSerializer(
+        help_text="Organisation information for the application",
+        read_only=False,
+        allow_null=True,
+        required=False,
+    )
+
+    application_round_pk = IntegerPrimaryKeyField(
+        queryset=ApplicationRound.objects.all(),
+        source="application_round",
+        help_text="Id of the application period for which this application is targeted to",
+    )
+
+    user = serializers.HiddenField(default=NullableCurrentUserDefault())
+
+    application_events = ApplicationEventCreateSerializer(
+        help_text="List of applications events", many=True
+    )
+
+    status = ChoiceCharField(
+        help_text="Status of this application", choices=ApplicationStatus.STATUS_CHOICES
+    )
+
+    billing_address = AddressCreateSerializer(
+        help_text="Billing address for the application",
+        allow_null=True,
+    )
+
+    applicant_type = serializers.CharField(allow_null=True)
+
+    home_city_pk = IntegerPrimaryKeyField(
+        queryset=City.objects.all(), source="home_city", required=False, allow_null=True
+    )
+
+    class Meta:
+        model = Application
+        fields = [
+            "pk",
+            "applicant_type",
+            "applicant_name",
+            "applicant_email",
+            "organisation",
+            "application_round_pk",
+            "contact_person",
+            "user",
+            "application_events",
+            "status",
+            "billing_address",
+            "home_city_pk",
+            "created_date",
+            "last_modified_date",
+            "additional_information",
+        ]
+
+    @staticmethod
+    def handle_person(contact_person_data: Dict[Any, Any]) -> Union[Person, None]:
+        if contact_person_data is not None:
+            # CASE: Create new person
+            if "pk" not in contact_person_data or contact_person_data["pk"] is None:
+                return PersonSerializer(data=contact_person_data).create(
+                    validated_data=contact_person_data
+                )
+
+            # CASE: Update existing person
+            return PersonSerializer(data=contact_person_data).update(
+                instance=Person.objects.get(pk=contact_person_data["pk"]),
+                validated_data=contact_person_data,
+            )
+
+        return None
+
+    def handle_organisation(
+        self, organisation_data: Dict[Any, Any]
+    ) -> Union[Organisation, None]:
+        if organisation_data is not None:
+            # CASE: Create new organisation
+            if "pk" not in organisation_data or organisation_data["pk"] is None:
+                return OrganisationSerializer(data=organisation_data).create(
+                    validated_data=organisation_data
+                )
+
+            # CASE: Update exisitng organisation
```

Comment:
```suggestion
            # CASE: Update existing organisation
```

Reply:
Nice found! Didn't see that even my IDE seems to nag about it :joy:

Reply:
fix'd